packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -1302,9 +1302,11 @@ mixin WidgetInspectorService {
       id = 'inspector-$_nextId';
       _nextId += 1;
       _objectToId[object] = id;
-      referenceData = _InspectorReferenceData(object);
-      _idToReferenceData[id] = referenceData;
-      group.add(referenceData);
+      if (object is DiagnosticsNode) {
```

Comment:
this is dropping everything on the floor that isn't a diagnostics node. I would expect the inspector in DevTools will be broken given this change.
The alternative design would be to require the object passed to `toId` to be a DiagnosticsNode (might be possible) or consider making the object held onto by _InspectorReferenceData be weak (if it isn't a String, num, bool) rather than making the change at the granularity of DiagnosticsNode.